sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkSecurityRuleImpl.java

```diff
@@ -294,6 +311,21 @@ public NetworkSecurityRuleImpl withDestinationApplicationSecurityGroup(String id
         return this;
     }
 
+    @Override
+    public NetworkSecurityRuleImpl withoutDestinationApplicationSecurityGroup(String id) {
+        destinationAsgs.remove(id);
+        return this;
+    }
+
+    @Override
+    public NetworkSecurityRuleImpl withDestinationApplicationSecurityGroup(String... ids) {
+        destinationAsgs = Arrays.stream(ids)
+            .collect(Collectors.toMap(Function.identity(), id -> new ApplicationSecurityGroupInner().withId(id)));
+        innerModel().withDestinationAddressPrefix(null);
+        innerModel().withDestinationAddressPrefixes(null);
+        return this;
+    }
```

Comment:
You add a replace scenario. How about append scenario? Should we consider it?

Reply:
In update flow, one can call multiple `withDestinationApplicationSecurityGroup` and `withoutDestinationApplicationSecurityGroup` to update the list.